src/commit.c

```diff
@@ -350,7 +350,11 @@ int apk_solver_commit_changeset(struct apk_database *db,
 
 	if (!db->performing_self_upgrade) {
 		if (errors)
-			snprintf(buf, sizeof(buf), "%d errors;", errors);
+			if (errors != 1)
+				snprintf(buf, sizeof(buf), "%d errors;", errors);
+			else
+				snprintf(buf, sizeof(buf), "%d error;", errors);
```

Comment:
I don't care about the generated code, but about the written code (duplication). ;)

I'd personally prefer simpler code, i.e. `%d error(s);` as you suggested.